src/main/java/bio/terra/cli/command/workspace/BreakGlass.java

```diff
@@ -0,0 +1,154 @@
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+/**
+ * This class corresponds to the third-level "terra workspace break-glass" command. This command is
+ * hidden in the usage help.
+ */
+@Command(
+    name = "break-glass",
+    description = "Grant break-glass access to a workspace user.",
+    hidden = true)
+public class BreakGlass extends BaseCommand {
+  private static final Logger logger = LoggerFactory.getLogger(BreakGlass.class);
+
+  @CommandLine.Option(
+      names = "--email",
+      required = true,
+      description = "Email of workspace user requesting break-glass access.")
+  private String granteeEmail;
+
+  @CommandLine.Option(
+      names = "--user-project-admin-sa",
+      required = true,
+      description =
+          "Path to the key file for a SA that has permission to set IAM policy on workspace projects for the current server.")
+  private String userProjectAdminSAKeyFile;
+
+  @CommandLine.Option(
+      names = "--big-query-sa",
+      required = true,
+      description =
+          "Path to the key file for a SA that has permissions to update the BigQuery dataset tracking break-glass requests.")
+  private String bigQuerySAKeyFile;
+
+  @CommandLine.Option(
+      names = "--big-query-project",
+      required = true,
+      description = "Project ID that contains the BigQuery dataset tracking break-glass requests.")
+  private String bigQueryProjectId;
+
+  @CommandLine.Option(
+      names = "--notes",
+      description = "Free text string about this request, to store in the BigQuery dataset.")
+  private String notes;
+
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+
+  /** Grant break-glass access to the workspace. */
+  @Override
+  protected void execute() {
+    workspaceOption.overrideIfSpecified();
+
+    // check that the SA key files exist and are valid
+    ServiceAccountCredentials userProjectsAdminCredentials;
+    ServiceAccountCredentials bigQueryCredentials;
+    try {
+      final List<String> SA_SCOPES =
+          ImmutableList.of("https://www.googleapis.com/auth/cloud-platform");
+      userProjectsAdminCredentials =
+          GoogleOauth.getServiceAccountCredential(
+              Path.of(userProjectAdminSAKeyFile).toFile(), SA_SCOPES);
+      bigQueryCredentials =
+          GoogleOauth.getServiceAccountCredential(Path.of(bigQuerySAKeyFile).toFile(), SA_SCOPES);
+    } catch (IOException ioEx) {
+      throw new UserActionableException("Error reading break-glass SA key files.", ioEx);
+    }
+
+    // grant the user's proxy group the Editor role on the workspace project
+    String granteeProxyGroupEmail =
+        Context.requireWorkspace().grantBreakGlass(granteeEmail, userProjectsAdminCredentials);
+
```

Comment:
I assume that if granting fails we'll throw out here?  It might be good to log failed attempts too?

Reply:
Yes, this will throw an exception if it fails and all exceptions are written to the CLI logs. Or do you mean to log it in the BQ dataset?

Reply:
Yes, it would probably be good to capture failed attempts in BQ dataset if it's not too much effort.

Reply:
It wouldn't be too much effort, but I guess I'm not sure the benefits of logging client-side CLI errors in a central location. A failed attempt to grant break-glass means that:
  - the requestor is not a workspace owner, in which case we'd tell them to request owner access and then re-run the command to log the grant
  - the script/command is broken (e.g. the SA lost folder permissions somehow), and then we need to go debug it

Maybe I'm looking at this wrong though -- how are you thinking we'd use the central record of failed attempts?

Reply:
I was just thinking that with the pen test coming up, someone might try to use this command as an attack vector to elevate privilege.   "Requestor not being a workspace owner" might be something we want an audit trail for from this perspective.  Agree there's not much value in logging basic user error or bugs in the code.

Reply:
That makes sense. I'll add a failure field to the BQ dataset schema and update it if the command fails because the user isn't a workspace owner.